### PR TITLE
Mitigate MSB4011 warning for importing ExperimentalFeatures.props 

### PR DESCRIPTION
**Pull request by jonthysell**
## Description

This PR fixes the `MSB4011` warning when `JSEngine.props` tries to import `ExperimentalFeatures.props` a second time. The expectation is that all RNW projects (app or module) import the `ExperimentalFeatures.props` file (located next to the VS solution) at the very top of each project file, so all RNW-related projects are in sync with respect to the state of any Experimental Features.

However, older RNW apps/module templates did not include this import, hence `JSEngine.props`'s workaround to import it. Unfortunately, MSBuild:

* Doesn't like project files to be imported twice, so it gives the `MSB4011` warning
* Doesn't provide any built-in way to detect if a project file has already been imported
* Doesn't allow you to suppress this particular warning completely

The best solution is to set a property in the imported file and check for that property before attempting to import the file.

This PR adds an `ReactExperimentalFeaturesSet` property to all in-repo and new template `ExperimentalFeatures.props` files, and changes `JSEngine.props` to both check for it before importing and throw a new warning so let existing projects know about the new property and how to resolve all of the warnings.

TLDR: After this PR has been merged, adding

```xml
<ReactExperimentalFeaturesSet>true</ReactExperimentalFeaturesSet>
```

to your `ExperimentalFeatures.props` file will get rid these annoying MSB4011 warnings.

### Type of Change
- Bug fix (non-breaking change which fixes an issue)

### Why

Honestly, I'm tired of this warning spamming my CI logs and obfuscating real issues.

Resolves my sanity

### What
See above.

## Screenshots
N/A

## Testing
Verified warning no longer appears for in-repo apps, and removing the property causes both the old and new warnings to appear.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10767)

### CODE REVIEW

**Comment by jonthysell** [comment]
Note, the warnings aren't harmful (MSBuild simply doesn't do the 2nd import), just annoying.

File: packages/e2e-test-app/windows/ExperimentalFeatures.props [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
   <PropertyGroup Label="Microsoft.ReactNative Experimental Features">
     <UseHermes>false</UseHermes>
+    
+    <ExperimentalFeaturesSet>true</ExperimentalFeaturesSet>
```
Comment (vmoroz) [inline packages/e2e-test-app/windows/ExperimentalFeatures.props:7]: >ExperimentalFeaturesSet [](http://example.com/codeflow?start=5&length=23)

Could we prefix it with "React"?
All these properties are global. We better to have our project-specific prefixes for all our properties. This is considered to be a good practice for MSBuild code. E.g.C++/WinRT code uses the CppWinRT prefix and their properties look like CppWinRTNamespaceMergeDepth, etc.

File: vnext/PropertySheets/JSEngine.props [hunk 2]
```diff
@@ -1,7 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
-  <Import Project="$(SolutionDir)\ExperimentalFeatures.props" Condition="Exists('$(SolutionDir)\ExperimentalFeatures.props')" />
+  <Target Name="EnsureExperimentalFeaturesSetTarget" BeforeTargets="PrepareForBuild" Condition="Exists('$(SolutionDir)\ExperimentalFeatures.props') And '$(ExperimentalFeaturesSet)' != 'true'">
+    <Warning Text="Property 'ExperimentalFeaturesSet' not set. Please specify &lt;ExperimentalFeaturesSet&gt;true&lt;/ExperimentalFeaturesSet>&gt; in '$(SolutionDir)\ExperimentalFeatures.props' to prevent the MSB4011 warnings." />
+  </Target>
+
+  <ImportGroup Condition="'$(ExperimentalFeaturesSet)' != 'true'">
```
Comment (vmoroz) [inline vnext/PropertySheets/JSEngine.props:8]: >Condition="'$(ExperimentalFeaturesSet)' != 'true'" [](http://example.com/codeflow?start=15&length=50)

Could we just add it to the condition on the Import tag below?

**Review by vmoroz — APPROVED** [review]
:shipit:
